src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2236,7 +2233,15 @@ private bool IsTypeDeclarationStart()
                 case SyntaxKind.ExplicitKeyword or SyntaxKind.ImplicitKeyword
                     when this.PeekToken(1).ContextualKind is SyntaxKind.ExtensionKeyword:
 
-                    return IsFeatureEnabled(MessageID.IDS_FeatureExtensions);
+                    // If we have `implicit extension E` then this is definitely an extension as it could not be parsed
+                    // as anything else.
+                    if (IsTrueIdentifier(this.PeekToken(2)))
+                        return true;
+
+                    // we have `implicit extension ...` technically this could be something like `implicit
+                    // extension.operator X(` (an implicit conversion with a explicit interface impl name). For compat,
+                    // determine the parsing strategy based on the lang version.
+                    return IsCompatBreakingFeatureEnabled(MessageID.IDS_FeatureExtensions);
```

Comment:
>return IsCompatBreakingFeatureEnabled(MessageID.IDS_FeatureExtensions); [](http://example.com/codeflow?start=20&length=71)

As I mentioned in other similar place. I am not comfortable with language version based approach.

Reply:
sgtm.